File: 237_DeleteNodeInALinkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        if(node && node -> next){
			ListNode *pre(NULL);
			while(node -> next){
				node -> val = node -> next -> val;
				pre = node;
				node = node -> next;
			}
			pre -> next = NULL;
		}
    }
};

void deleteNode(ListNode* node) {
    *node = *node->next;
}

void deleteNode(ListNode* node) {
    auto next = node->next;
    *node = *next;
    delete next;
}

File: 237_DeleteNodeInALinkedList.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        if node and node.next:
            while node.next:
                node.val = node.next.val
                pre, node = node, node.next
            pre.next = None

def deleteNode(self, node):
    node.val = node.next.val
    node.next = node.next.next
